### PR TITLE
Budget-App

### PR DESCRIPTION
**Pull request by deniodev**
**Interactions**

- Splash screen
  - A simple page with the name of your app, and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - The button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

**Testing**

- Created unit and integration tests for all the most important components of your RoR application.
Important technical concepts

- Used Postgres as your database.

- Used devise for authentication.

- Validated all user input to make sure that anyone with bad intentions cannot compromise your app.

- Used a view template engine (.erb).

- The project is deployed and accessible online via this [link]( https://budget-buddy-yniy.onrender.com/)

My database schema should reflect the following [structure:](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/images/erd_diagram.png)

Project documentation
Here is a video [presentation](https://www.loom.com/share/Budget-Buddy-Web-App-Presentation-f8255e0f34824136babc3caed1aade38?sid=15571058-2deb-44c0-8118-12f07f2281e0)

I used a design by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,180 @@
...
+
+* I am grateful to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) for the priviledge to use his design.
+
+* I would like to thank my Microverse team and my coding partners for their support and guidance.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## Presentation <a name="presentation"></a>
+
+* [video](https://www.loom.com/share/Budget-Body-Web-App-Presentation-2def64cc00ed494ba3b785f0693beef6?sid=788bbade-d5f7-40db-886c-7c060c8ff33e) presentation.
```
Comment (nedjwab) [inline README.md]: [ ] your video presentation should contain a highlight about a piece of the code you find interesting or is proud of making, kindly fix that 🤓.

File: app/views/items/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<%= turbo_frame_tag "new_transaction" do %>
```
Comment (nedjwab) [inline app/views/items/new.html.erb:1]: [ ] At the top of the page the user can see the total amount for the category (sum of all of the amounts of the transactions in that category), kindly fix that 🤓.

your transaction page: 👇
 
transaction

how it should be like 👇 : 

design

**Review by nedjwab — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @deniodev    

Nice job, you are almost there.



## (Highlights) Good Points: :
- Good implementation of some of the project requirements  ✔
- GitHub flow was correctly used ✔.
- Your design is  🔥🔥🔥🔥

## (Changes Required) Aspects to improve: ♻️
You did great implementing some of the project requirements, but you still have some issues: 
check the comments below 
 👇👇👇

 
## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Munsa1 — APPROVED** [review]
Hi @deniodev, 
congrats 2

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**